merge/wipe_fields_test.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package merge_test
+
+import (
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+	. "sigs.k8s.io/structured-merge-diff/v4/internal/fixture"
+	"sigs.k8s.io/structured-merge-diff/v4/merge"
+	"sigs.k8s.io/structured-merge-diff/v4/typed"
+)
+
+func TestWipeManagedFields(t *testing.T) {
+	type testCase struct {
```

Comment:
Any chance we can use the testing mechanism we already have to test these? That'd help build more sophisticated scenarios, and understand how these situation happen. I can't quite see how to represent the "prepare" in there though.

Reply:
I looked at it, but the way it works right now I couldn't see a good way of adding the prepared stuff without making the tests much more complicated :-/